src/lex.cpp

```diff
@@ -450,10 +450,34 @@ void lexer::parse_current_token() {
         }
 
       case '\n':
-      case '\r':
+      case '\r': {
+        char8* c_up_to_first_matching_quote = nullptr;
+        char8* c_end = c;
```

Comment:
Hmm, do you think this code would be better in its own function? In a separate function, better variable names might come more naturally.